### PR TITLE
Feature/form uix updates

### LINKED ISSUE #32: Display success message on successful form submission
Display a dialog on successful host / refugee form submittal and redirect to home page on dialog close.

### LINKED ISSUE #36: Update form labels based on host / refugee selection
- [ ] Country
- [ ] Number of People

### LINKED ISSUE #37: Refugee / host required for form submittal
Refugee / host field is required but the form is able to be submitted without selecting an option.

### PR DESCRIPTION
**Pull request by Seth-McKilla**
Closes #32
Closes #36
Closes #37 

https://www.loom.com/share/381e612cd85c44d58463db97850cb20b

### CODE REVIEW

File: components/signupForm.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ export const SignupForm = () => {
       userType: (value) => !!value,
       phoneNumber: (value) => {
         var regEx = `^\\+?\\(?([0-9]{1,4})\\)?([-. ]?([0-9]{2}))?([-. ]?([0-9]{3}))([-. ]?([0-9]{2,3}))([-. ]?([0-9]{2,4}))$`;
-        return value.match(regEx) !== null;
+        return !!value || value.match(regEx) !== null;
```
Comment (danielavelez1201) [inline components/signupForm.tsx:63]: good catch!

File: components/signupForm.tsx [hunk 2]
```diff
@@ -48,40 +54,54 @@ export const SignupForm = () => {
...
   };
+
   return (
     <Paper padding="lg" shadow="sm" radius="md" withBorder>
       <Title order={3}>Register</Title>
       <Space h="lg" />
 
       <form onSubmit={form.onSubmit(onSubmitHandler)}>
         <Group grow direction="column">
+          <LoadingOverlay visible={isSubmitting} />
```
Comment (danielavelez1201) [inline components/signupForm.tsx:125]: this is great!

File: components/signupForm.tsx [hunk 3]
```diff
@@ -108,10 +112,12 @@ export const SignupForm = () => {
...
 
       <Modal
-        opened={true}
+        opened={isSuccess}
         onClose={handleModalClose}
         title="Success!"
-        message="You have successfully registered. We're working on finding you a ___ and will be in touch with you as soon as possible!"
+        message={`Your registration was successful. We're working on matching you with a ${
+          userType === 'refugee' ? 'host' : 'refugee'
+        } and will be in touch with you as soon as possible!`}
```
Comment (danielavelez1201) [inline components/signupForm.tsx:120]: looks great! if the user type is a "host," I would also include a "Thank you for your support in these tough times" or something like that to show appreciation

File: components/signupForm.tsx [hunk 4]
```diff
@@ -181,14 +195,20 @@ export const SignupForm = () => {
             defaultValue={2}
             {...form.getInputProps('groupSize')}
             placeholder="Number of people"
-            label="Number of people"
+            label={
+              userType === 'refugee'
+                ? 'Number of people in your group'
+                : 'Number of people you can accomodate'
```
Comment (danielavelez1201) [inline components/signupForm.tsx:201]: something to look into might be if we have to collect information for all the refugees in the group to share with the organization, so then choosing a number > 1 could open up more fields to enter the others' info

**Review by danielavelez1201 — APPROVED** [review]
looks awesome!

File: components/signupForm.tsx (hunk 3 above)
Comment (Seth-McKilla) [reply components/signupForm.tsx:120]: I really like this, good idea! Definitely going to add that before merging with main.

File: components/signupForm.tsx (hunk 4 above)
Comment (Seth-McKilla) [reply components/signupForm.tsx:201]: Definitely a good thought 🤔 something we should keep in mind once we start to develop relationships with local organizations, can morph the form into something that gives them exactly what they're looking for.

**Pull request merged**